fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1563,59 +1563,120 @@ public void replayAddPartition(PartitionPersistInfo info) throws DdlException {
             throw InvalidOlapTableStateException.of(olapTable.getState(), olapTable.getName());
         }
-
-        if (!olapTable.checkPartitionNameExist(partitionName, isTempPartition)) {
-            if (clause.isSetIfExists()) {
-                LOG.info("drop partition[{}] which does not exist", partitionName);
-                return;
-            } else {
-                ErrorReport.reportDdlException(ErrorCode.ERR_DROP_PARTITION_NON_EXISTENT, partitionName);
-            }
-        }
-
-        PartitionInfo partitionInfo = olapTable.getPartitionInfo();
         if (!partitionInfo.isRangePartition() && partitionInfo.getType() != PartitionType.LIST) {
             throw new DdlException("Alter table [" + olapTable.getName() + "] failed. Not a partitioned table");
         }
+        List<String> partitionNames = new ArrayList<>();
+        boolean isTempPartition = clause.isTempPartition();
 
-        // drop
-        if (isTempPartition) {
-            olapTable.dropTempPartition(partitionName, true);
-        } else {
-            Partition partition = olapTable.getPartition(partitionName);
-            if (!clause.isForceDrop()) {
-                if (partition != null) {
-                    if (stateMgr.getGlobalTransactionMgr()
-                            .existCommittedTxns(db.getId(), olapTable.getId(), partition.getId())) {
-                        throw new DdlException(
-                                "There are still some transactions in the COMMITTED state waiting to be completed." +
-                                        " The partition [" + partitionName +
-                                        "] cannot be dropped. If you want to forcibly drop(cannot be recovered)," +
-                                        " please use \"DROP PARTITION <partition> FORCE\".");
+        if (clause.hasMultiPartitions()) {
+            PartitionDesc partitionDesc = clause.getPartitionDesc();
+            if (partitionDesc instanceof MultiRangePartitionDesc) {
+
+                if (!(partitionInfo instanceof RangePartitionInfo)) {
+                    throw new DdlException("Batch deletion of partitions only support range partition tables.");
+                }
+                RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+
+                List<Column> partitionColumns = rangePartitionInfo.getPartitionColumns();
+                if (partitionColumns.size() != 1) {
+                    throw new DdlException("Alter batch drop partition only support single range column.");
+                }
+
+                Column firstPartitionColumn = partitionColumns.get(0);
+                MultiRangePartitionDesc multiRangePartitionDesc = (MultiRangePartitionDesc) partitionDesc;
+                boolean isAutoPartitionTable = false;
+                if (partitionInfo.getType() == PartitionType.EXPR_RANGE) {
+                    isAutoPartitionTable = true;
```

Comment:
If it's clear for every kind of partition type, `isAutoPartition` should be a {const} field/method of the PartitionInfo.
cc @nshangyiming